### PR TITLE
Docs: New Explanation sub-levels + high-level introduction as explanation (Diátaxis)

### PR DESCRIPTION
**Pull request by benjaoming**
Reviewing, cc: @ericholscher 

* The main main main contribution here is to organize the explanation section and add a very important main introduction, "Choosing a dedicated documentation platform".

* Currently, this will have a lot of text that can be shortened or clarified.

* You might also want to add something, but please see the comments at the top of "Choosing a dedicated documentation platform"

* A lot of intentions are left behind as reST comments.. they can be eliminated by simply choosing to "not do this" and delete the comment.

Supersedes https://github.com/readthedocs/readthedocs.org/pull/10063/files with the full context of a new Explanation section.

Some filenames are now so badly misleading that they should be changed.

* [x] integrations.rst => explanation/continuous-documentation.rst + redirect
* [x] choosing-a-site.rst => explanation/dedicated-platform.rst + redirect
* [x] ~Possibly, add a graphic for "Continuous Documentation" to replace the lengthy Webhook explanation~ just remove that explanation for now

---
:books: Documentation previews :books:

- User's documentation (`docs`): https://docs--10071.org.readthedocs.build/en/10071/

- Developer's documentation (`dev`): https://dev--10071.org.readthedocs.build/en/10071/

### CODE REVIEW

**Comment by benjaoming** [comment]
Btw... since there's a lot of new content, maybe the best way to start reviewing it is to build it locally since the PR preview has a lot of disturbing "diff view" :)

**Comment by ericholscher** [comment]
@benjaoming I wonder about breaking out the structure changes here from the content into diff PR's? That way we can merge the index changes, and work on each of the pages in dedicated PR's that unblocks each of them?

**Comment by benjaoming** [comment]
Just looking at it quickly, the easy part is to put "Deep dive into Read the Docs" into a separate PR and leave the rest in this PR?

**Comment by benjaoming** [comment]
The methodology part can also be in a separate PR, just maybe have to remove some references to it.

**Comment by benjaoming** [comment]
It will add extra work, so if these changes can be reviewed here, it's certainly worth a shot.

**Comment by benjaoming** [comment]
Uncovered another inconvenience of running a fork... chained PRs are impossible. Anyways, this PR was opened before I stopped the pattern, so please :bear: with me :)

File: docs/user/environment-variables.rst [hunk 1]
```diff
@@ -4,11 +4,11 @@ Understanding environment variables
 ===================================
 
 Read the Docs allows you to define your own environment variables to be used in the build process.
-It also defines a set of :doc:`default environment variables </reference/environment-variables>` with information about your build.
+It also defines a set of :doc:`pre-defined environment variables </reference/environment-variables>` with information about your build.
```
Comment (benjaoming) [inline docs/user/environment-variables.rst:7]: There might be more of these, I haven't checked. This is beyond what I can remember, but at some point we started saying "pre-defined environment variables", I'm guessing that it's less ambiguous and much easier to understand when put up next to "user-defined environment variables".

**Comment by benjaoming** [comment]
@ericholscher methodology is now split out in another PR which has reduced the size of this PR and review work :+1: 

I was contemplating to split out "Deep Dive into Read the Docs", too.. but it's so little.

I'll take a quick tour again to see what improvements I can uncover, but otherwise it's ready for review :+1:

File: docs/user/explanation/continuous-documentation.rst [hunk 2]
```diff
@@ -7,28 +7,50 @@
...
 
-Read the Docs is a *Continuous Documentation Deployment* platform for your software project.
-Every time you change something in your documentation, Read the Docs will detect your change and build your documentation.
+Read the Docs is a *Continuous Documentation* platform for your software project.
+Every time you change something in your documentation, Read the Docs will automatically detect your changes,
+*build* your documentation,
+and finally *deploy* your documentation for readers to see.
 
 The Continuous Integration and Continuous Deployment (CI/CD) features are configured with your repository provider,
 such as GitHub, Bitbucket or GitLab.
```
Comment (humitos) [inline docs/user/explanation/continuous-documentation.rst:19]: we could use the `|or_|` provider sustitution here.

File: docs/user/explanation/dedicated-platform.rst [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+🔋️ :ref:`batteries_included`
+    Read the Docs continues to develop new projects and ideas,
+    bringing additional powers to documentation projects that are hosted on the platform.
+
+.. The following sections are about using a dedicated platform as a first-choice in order to free up space for other tasks.
+.. These paragraphs can benefit from being shortened and organized, but they are intended as the summary of the introduction.
+
+Using Read the Docs
+allows you to focus on other critical tasks,
+such as choosing documentation tools, structuring your documentation and having the perfect workflow.
```
Comment (humitos) [inline docs/user/explanation/dedicated-platform.rst:45]: "Structuring your documentation" could highlight/mention the page where we talk about Diataxis that I think we already wrote.

**Review by humitos — COMMENTED** [review]
I'm not in my best day to do a review of this content. However, I took a quick look at it and made some initial comments.

The goal/idea is great, in my opinion, and I think we really want/need this page with narrative content that newcomers can come and read as their first contact to Read the Docs.

I haven't reviewed the main content, tho, but I think it goes in the right direction as a whole.

File: docs/user/explanation/continuous-documentation.rst [hunk 4]
```diff
@@ -7,28 +7,50 @@
...
+========================
 
-Read the Docs is a *Continuous Documentation Deployment* platform for your software project.
-Every time you change something in your documentation, Read the Docs will detect your change and build your documentation.
+Read the Docs is a *Continuous Documentation* platform for your software project.
+Every time you change something in your documentation, Read the Docs will automatically detect your changes,
+*build* your documentation,
+and finally *deploy* your documentation for readers to see.
 
 The Continuous Integration and Continuous Deployment (CI/CD) features are configured with your repository provider,
```
[the remark this answers is not in the document]
Comment (benjaoming) [reply docs/user/explanation/continuous-documentation.rst:18]: Yes, haha, I added it in the glossary just today, having a glossary with pop-up tooltips enabled makes writing much easier :100:

File: docs/user/explanation/continuous-documentation.rst (hunk 2 above)
Comment (benjaoming) [reply docs/user/explanation/continuous-documentation.rst:19]:
```suggestion
such as |git_providers_or|.
```

File: docs/user/explanation/dedicated-platform.rst (hunk 3 above)
Comment (benjaoming) [reply docs/user/explanation/dedicated-platform.rst:45]: Yes, it definitely should. This content was just recently broken off into a separate PR to make this one lighter :)

File: docs/user/glossary.rst [hunk 5]
```diff
@@ -39,6 +44,14 @@ so that you have a reference for how we're using them.
       and rules for publication of documentation can be :doc:`automated </automation-rules>`.
       Similar to :term:`Docs as Code`.
 
+   webhooks
+      There are two important types of webhooks for Read the Docs:
```
[the remark this answers is not in the document]
Comment (benjaoming) [reply docs/user/glossary.rst]: Sounds good! I gave it a little tweak :)

File: docs/user/glossary.rst [hunk 6]
```diff
@@ -59,6 +72,18 @@ so that you have a reference for how we're using them.
       and hyphens. You can retrieve your project or version slugs from
       :doc:`our API <api/v3>`.
 
+   static website
+      Documentation projects served on Read the Docs are a type of *static websites*.
+
+      A static site or static website is a collection of HTML files, images, CSS and JavaScript that do not change over time,
+      nor depend on which web server that is serving them.
+
```
[the remark this answers is not in the document]
Comment (benjaoming) [reply docs/user/glossary.rst]: Sounds good! I gave it a little tweak :)

File: docs/user/explanation/dedicated-platform.rst [hunk 7]
```diff
@@ -0,0 +1,317 @@
...
+a dedicated documentation platform is a reliable first choice.
+
+A dedicated platform helps to solve tasks that you otherwise end up writing and maintaining your own scripts and CI tools for.
+
+.. seealso::
+
+    `Comparison with GitHub Pages <https://about.readthedocs.com/comparisons/github-pages/>`__
+        On our website, we have a list of common tasks that developers and DevOps teams have to solve themselves on a generic CI.
+
+Simply put, the role of a *dedicated documentation platform* is to offer a compelling set of features that documentation projects and their organizations or communities need.
```
[the remark this answers is not in the document]
Comment (benjaoming) [reply docs/user/explanation/dedicated-platform.rst]: I absolutely agree. It was very easy to write this sentence after writing all the other stuff, but it sounds better in the beginning. The intro paragraph was also kind of weak, so this made it better :+1:

File: docs/user/explanation/dedicated-platform.rst (hunk 3 above)
Comment (benjaoming) [reply docs/user/explanation/dedicated-platform.rst:47]: Added a note :)

File: docs/user/explanation/continuous-documentation.rst [hunk 8]
```diff
@@ -7,28 +7,50 @@
...
-When a receive a *webhook* notification, we match it to a project's *Integration*.
-When a webhook is received, the matching project will then:
+When a receive a *webhook* notification, we match it to a Read the Docs project.
+The matching project will then process your build and publish the documentation.
 
-* :doc:`Build </builds>` the latest commit.
-* Synchronize your versions based on the latest tag and branch data in Git.
-* Run your :doc:`automation rules</automation-rules>`.
-* Auto-cancel any currently running builds of the same version.
-* Add a log entry to the integration's :guilabel:`Recent Activity`.
```
Comment (ericholscher) [inline docs/user/explanation/continuous-documentation.rst:27]: I'm a little confused how we keep migrating this page away from actually documenting the workflow for integrations. Do we have a page in the docs that explains how to setup GitHub on your repo, and what that flow looks like?

File: docs/user/index.rst [hunk 9]
```diff
@@ -123,11 +120,11 @@ Get a high-level overview of our platform:
...
-  Discover the advantages of having your documentation continuously deployed.
+💡 :doc:`/explanation/dedicated-platform`
+  An introduction to the reasons behind choosing a dedicated documentation platform such as Read the Docs.
 
-💡 :doc:`/choosing-a-site`
-  Learn about the differences between |org_brand| and |com_brand|.
+💡 :doc:`/explanation/continuous-documentation`
+  Discover the advantages of having your documentation continuously deployed.
 
 .. TODO: This next item needs its article to be finished in a separate PR
```
Comment (ericholscher) [inline docs/user/index.rst:129]: I thought the goal of this PR was to finish the deep dive, and add it to the index?

**Review by ericholscher — COMMENTED** [review]
Overall this is a great change, but I worry that we're confusing the value we're providing to the user in terms of explanation, and not actually documenting how things work at a technical level, which is something the old docs did well around integrations that seem to be lost.

File: docs/user/index.rst (hunk 9 above)
Comment (benjaoming) [reply docs/user/index.rst]: Thanks for fixing!

File: docs/user/explanation/continuous-documentation.rst (hunk 8 above)
Comment (benjaoming) [reply docs/user/explanation/continuous-documentation.rst:27]: Changes to add a feature for Git provider integrations are now parked here https://github.com/benjaoming/readthedocs.org/pull/5

I'll move them once this PR is merged.

**Comment by ericholscher** [comment]
Going to close this, and pull the small changes into a future PR.

**Pull request closed without merging**